src/verinfast/agent.py

```diff
@@ -467,6 +493,9 @@ def parseRepo(self, path: str, repo_name: str):
             if not self.config.dry:
                 dependencies_output_file = dependency_walk(output_file=dependencies_output_file, logger=self.log)
             self.log(msg=dependencies_output_file, tag="Dependency File", display=False)
+            if not self.config.dry:
```

Comment:
why don't you move this check into the function so it's not called a bunch of times?